pkg/apis/allocation/v1/gameserverallocation_test.go

```diff
@@ -1016,6 +1016,231 @@ func TestGameServerSelectorMatches(t *testing.T) {
 	}
 }
 
+// Helper function for creating int64 pointers
+func int64Pointer(x int64) *int64 {
+	return &x
+}
+
+func TestGameServerCounterActions(t *testing.T) {
```

Comment:
nit: it would be great if we can assert the returned error is expected for some of the unhappy case.

Reply:
Right now I'm checking if it throws (or does not throw) an error. Are you recommending testing the error message is as expected?

Reply:
Yes, that's what I meant. Not a blocker though.

Reply:
General guidance is to not check error strings (otherwise the test becomes a change detector), and instead to use separate error types like we have in pkg/apis/agones/v1/common.go and check on error type. Creating specific error types seems a bit overkill for this particular case, but I could certainly do it if you think it's worth the improvement to the unit tests?